src/base/utils/fs.cpp

```diff
@@ -347,15 +347,9 @@ bool Utils::Fs::isNetworkFileSystem(const QString &path)
         file += '/';
     file += '.';
 
-#if defined(Q_OS_MACOS)
-    struct statfs64 buf {};
```

Comment:
According to the doc you linked in the first post, we actually need `_DARWIN_FEATURE_64_BIT_INODE` to be defined (for `buf.f_type` to be useful). Can you confirm it is set?
Also https://stackoverflow.com/a/66031426, perhaps adding a compiler flag would be better.

Reply:
checked, this is already defined for both platforms (i.e. x86_64 and arm64). this define comes from some include file (don't know what exactly, but it does). I added the following lines before and after all includes in edited file:

```cpp
#if defined(_DARWIN_FEATURE_64_BIT_INODE)
#error "_DARWIN_FEATURE_64_BIT_INODE is defined"
#endif
```

and error appeared only in second case (i.e. after includes), so this is defined by some header file, not by compiler

so, I think no additional changes are required, removing deprecated function is enough. very likely Qt cares about it

Reply:
I still prefer qbt define the macro on its own instead of relying on 3rd party factors. I can do it in another PR.

Reply:
>I can do it in another PR.

#15661